src/polar/Pie.tsx

```diff
@@ -779,10 +753,149 @@ export class Pie extends PureComponent<InternalProps, State> {
         {label && this.renderLabels(sectors)}
         {Label.renderCallByParent(this.props, null, false)}
         {(!isAnimationActive || isAnimationFinished) && LabelList.renderCallByParent(this.props, sectors, false)}
-        <SetPolarGraphicalItem data={this.props.data} dataKey={this.props.dataKey} hide={this.props.hide} />
-        <SetTooltipEntrySettings fn={getTooltipEntrySettings} args={this.props} />
-        <SetPiePayloadLegend sectors={this.props.sectors} legendType={this.props.legendType} />
       </Layer>
     );
   }
 }
+
+const defaultPieProps: Partial<Props> = {
+  stroke: '#fff',
+  fill: '#808080',
+  legendType: 'rect',
+  cx: '50%',
+  cy: '50%',
+  startAngle: 0,
+  endAngle: 360,
+  innerRadius: 0,
+  outerRadius: '80%',
+  paddingAngle: 0,
+  labelLine: true,
+  hide: false,
+  minAngle: 0,
+  isAnimationActive: !Global.isSsr,
+  animationBegin: 400,
+  animationDuration: 1500,
+  animationEasing: 'ease',
+  nameKey: 'name',
+  blendStroke: false,
+  rootTabIndex: 0,
+};
+
+function PieImpl(props: Props) {
+  const cells = useMemo(() => findAllByType(props.children, Cell), [props.children]);
+  const updateId = useUpdateId();
+  const pieSettings: ResolvedPieSettings = useMemo(
+    () => ({
+      name: props.name,
+      nameKey: props.nameKey,
+      tooltipType: props.tooltipType,
+      data: props.data,
+      dataKey: props.dataKey,
+      cx: props.cx,
+      cy: props.cy,
+      startAngle: props.startAngle,
+      endAngle: props.endAngle,
+      minAngle: props.minAngle,
+      paddingAngle: props.paddingAngle,
+      innerRadius: props.innerRadius,
+      outerRadius: props.outerRadius,
+      cornerRadius: props.cornerRadius,
+    }),
+    [
+      props.cornerRadius,
+      props.cx,
+      props.cy,
+      props.data,
+      props.dataKey,
+      props.endAngle,
+      props.innerRadius,
+      props.minAngle,
+      props.name,
+      props.nameKey,
+      props.outerRadius,
+      props.paddingAngle,
+      props.startAngle,
+      props.tooltipType,
+    ],
+  );
+
+  const presentationProps = filterProps(props, false);
+
+  const sectorsAndCoord = useAppSelector(state => {
+    // @ts-expect-error result of filterProps is not PieProps, but is treated as such
+    return selectPieSectors(state, pieSettings, presentationProps, cells) ?? {};
+  });
+  const sectors = sectorsAndCoord?.sectors;
+
+  // this causes way too many re-renders
+  // useLegendPayloadDispatch(computeLegendPayloadFromPieData, { ...props, sectors });
```

Comment:
Any clue what causes the re-renders? Do the sectors depend on offset?

Reply:
Ah right yes it does. Well we _could_ compute the sectors in two steps, first just the data, then feed the data (plus name plus fill color) into the legend and sector position selectors, then feed the output of sector positions into Pie.

That, or break the API that was never documented anyway and compute Legend from raw data instead of sectors. I am fan of this option because that's what I wanted to do either way. Why should we expose internal representation of the payload and pretend it's given by the user?

Reply:
Yes, but I haven't quite figured out what's up there. I commented out where I pass offset in because it was causing tests to fail - I need to go back to current 3.x and see if offset is actually getting passed or not from the generator

Reply:
Agreed on the second option for the most part, was just trying to get away with the least work :p. 

The thing is, we should give the user the data (computed or otherwise) they need when they go make a custom legend. But it should be explicitly documented rather than just stuffing whatever is there into the "payload"